src/services/side-panels/side-panel-item-dto.ts

```diff
@@ -3,22 +3,35 @@ import {ui} from 'kaltura-player-js';
 import {ReservedPresetName, SidePanelPosition, SidePanelMode} from './types/types';
 const {ReservedPresetNames} = ui;
 
-export class SidePanelItemDto {
+export interface ISidePanelItemDto {
   readonly label: string;
   readonly renderIcon?: ComponentClass | FunctionalComponent;
   readonly renderContent: ComponentClass | FunctionalComponent;
   readonly presets?: ReservedPresetName[];
   readonly position: SidePanelPosition;
   readonly expandMode: SidePanelMode;
-  readonly hooks?: {onActivate: (() => void) | null; onDeactivate: (() => void) | null};
+  readonly onActivate?: () => void;
+  readonly onDeactivate?: () => void;
+}
+
+export class SidePanelItemDto implements ISidePanelItemDto {
```

Comment:
I think ISidePanelItem it's clearer

Reply:
Already fixed, the interface name too ?
i did it easily distinguish between internal type to the external  type used outside